src/test/java/com/mossle/cdn/CdnUtilsTest.java

```diff
@@ -0,0 +1,100 @@
+package com.mossle.cdn;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.UUID;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.net.ssl.*"})
+public class CdnUtilsTest {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @PrepareForTest({CdnUtils.class})
+    @Test
+    public void testCopyUrlToFile() throws Exception {
+        PowerMockito.mockStatic(UUID.class);
+        Date date = new Date(1560600794000L);
+        PowerMockito.whenNew(Date.class).withNoArguments().thenReturn(date);
+        UUID uUID = PowerMockito.mock(UUID.class);
+        PowerMockito.when(uUID.randomUUID().toString()).thenReturn("baz");
+
+        Assert.assertEquals("/2019/06/15/baz.uk/",
+                CdnUtils.copyUrlToFile(
+                        "foo", "https://www.bbc.co.uk/", "baz"));
+        Assert.assertEquals("/2019/06/15/baz.uk/",
+                CdnUtils.copyUrlToFile(
+                        "foo", "https://www.bbc.co.uk/", "baz", null));
```

Comment:
Does this test actually access the network? We definitely should not do that.

Reply:
Also, we shouldn't add in the URL of an entirely unrelated company without their permission...

Reply:
Yeah.. struggling to mock this bit.. Might have to remove the test

Reply:
Test Method replaced with ExpectedException